### PR TITLE
feat(Board Game Arena): add presence

### LINKED ISSUE #5389: Board Game Arena | boardgamearena.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4463

<div type='discussions-op-text'>

<sup>Originally posted by **bastien09** May  1, 2020</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months?
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [x] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** Calys#9504

**What's the name of the service?** https://boardgamearena.com/ | Board Game Arena

**What should the Presence display?** The game you're currently playing

**If possible, please provide a logo for the service (512x512 minimum)** Not exactly 512, but that's the highest resolution I found for their logo https://x.boardgamearena.net/data/newsimg/logo2016.png
</div>

### PR DESCRIPTION
**Pull request by theusaf**
## Description 

* Resolves #5389
* Adds details for all games
  * ~400 detailed
  * ~200 left as stubbed

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/B/Board Game Arena/games/C/conspiracy.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const conspiracy: GamePresence = {
+	logo: "https://i.imgur.com/zQCv2iy.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/C/conspiracy.ts:13]: Probs invisible on white themes

File: websites/B/Board Game Arena/games/C/crusadersthywillbedone.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+				smallImageText: `Score: ${getPlayerScore(userPlayer)}`,
+			};
+		if (activePlayer === userPlayer || gameStateType !== "activeplayer") {
+			switch (gameState) {
+				case "gameEnd":
+					data.state = "Viewing game results";
+					break;
+			}
+		} else data.state = `Waiting for ${activePlayerData.name}`;
+		return data;
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/C/crusadersthywillbedone.ts:31]: Add stub todo?

**Review by Bas950 — CHANGES_REQUESTED** [review]
Checked the main files and the A, B and C folders.
Will continue another day
image

File: websites/B/Board Game Arena/games/A/artthief.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const artthief: GamePresence = {
+	logo: "https://i.imgur.com/dmI5AxZ.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/A/artthief.ts:13]: image

File: websites/B/Board Game Arena/games/A/astra.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const astra: GamePresence = {
+	logo: "https://i.imgur.com/mjLDfXb.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/A/astra.ts:13]: image

File: websites/B/Board Game Arena/games/B/bigtimesoccer.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const bigtimesoccer: GamePresence = {
+	logo: "https://i.imgur.com/tHJtiHr.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/B/bigtimesoccer.ts:13]: image

File: websites/B/Board Game Arena/games/B/bigtwo.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const bigtwo: GamePresence = {
+	logo: "https://i.imgur.com/GUVbanU.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/B/bigtwo.ts:13]: image

File: websites/B/Board Game Arena/games/D/dronesvsseagulls.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const dronesvsseagulls: GamePresence = {
+	logo: "https://i.imgur.com/7VgKK13.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/D/dronesvsseagulls.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/D/dudo.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const dudo: GamePresence = {
+	logo: "https://i.imgur.com/7x5X289.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/D/dudo.ts:13]: dudo perudo? is this correct

**Review by Bas950 — CHANGES_REQUESTED** [review]
D Folder

File: websites/B/Board Game Arena/games/E/egocentricworld.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const egocentricworld: GamePresence = {
+	logo: "https://i.imgur.com/0KlskoO.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/E/egocentricworld.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/E/ekonos.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const ekonos: GamePresence = {
+	logo: "https://i.imgur.com/DHYdQ8g.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/E/ekonos.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/E/emdomicrocosm.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const emdomicrocosm: GamePresence = {
+	logo: "https://i.imgur.com/Bh8LLSW.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/E/emdomicrocosm.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/E/eminentdomain.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const eminentdomain: GamePresence = {
+	logo: "https://i.imgur.com/3UkS8Ot.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/E/eminentdomain.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/E/escapefromthehiddencastle.ts [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const escapefromthehiddencastle: GamePresence = {
+	logo: "https://i.imgur.com/D7Y5Gcm.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/E/escapefromthehiddencastle.ts:13]: Is this the correct logo?

**Review by Bas950 — CHANGES_REQUESTED** [review]
E Folder

File: websites/B/Board Game Arena/games/D/dudo.ts (hunk 8 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/D/dudo.ts:13]: Yes, this is correct. There are a lot of them like this where the id of the game doesn't seem to match the name on the logo.

File: websites/B/Board Game Arena/games/E/ekonos.ts (hunk 10 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/E/ekonos.ts:13]: image

File: websites/B/Board Game Arena/games/E/emdomicrocosm.ts (hunk 11 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/E/emdomicrocosm.ts:13]: image

File: websites/B/Board Game Arena/games/E/escapefromthehiddencastle.ts (hunk 13 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/E/escapefromthehiddencastle.ts:13]: I can confirm this is the correct logo. https://boardgamearena.com/gamepanel?game=escapefromthehiddencastle

File: websites/B/Board Game Arena/games/F/factum.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const factum: GamePresence = {
+	logo: "https://i.imgur.com/s7EppM0.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/F/factum.ts:13]: check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
F folder

**Review by Bas950 — CHANGES_REQUESTED** [review]
G folder

File: websites/B/Board Game Arena/games/H/herrlof.ts [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+			switch (gameState) {
+				case "bidding":
+					data.state = "Bidding";
+					break;
+				case "playCard":
+					data.state = "Playing a card";
+					break;
+				case "threeOptOut":
+				case "resolveThree":
+					data.state = "Using the 3 special ability";
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/H/herrlof.ts:34]: `abilities`?

**Review by Bas950 — CHANGES_REQUESTED** [review]
H, I, and J folders

File: websites/B/Board Game Arena/games/Y/yatzy.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+	getActivePlayerId,
+	getCurrentGameState,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const yatzy: GamePresence = {
+	logo: "https://i.imgur.com/cOG3G3P.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/Y/yatzy.ts:12]: check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
W, X, Y, and Z folders

File: websites/B/Board Game Arena/games/G/gangsta.ts [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const gangsta: GamePresence = {
+	logo: "https://i.imgur.com/NswRouz.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/G/gangsta.ts:13]: image

File: websites/B/Board Game Arena/games/G/gardennation.ts [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const gardennation: GamePresence = {
+	logo: "https://i.imgur.com/6K4YKPc.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/G/gardennation.ts:13]: image

File: websites/B/Board Game Arena/games/G/glow.ts [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const glow: GamePresence = {
+	logo: "https://i.imgur.com/kU8awMi.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/G/glow.ts:13]: image

File: websites/B/Board Game Arena/games/H/hex.ts [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const hex: GamePresence = {
+	logo: "https://i.imgur.com/60R27j7.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/H/hex.ts:13]: image

File: websites/B/Board Game Arena/games/Y/yatzy.ts (hunk 16 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/Y/yatzy.ts:12]: image

File: websites/B/Board Game Arena/games/G/goldwest.ts [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const goldwest: GamePresence = {
+	logo: "https://i.imgur.com/IHYvIls.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/G/goldwest.ts:13]: image

File: websites/B/Board Game Arena/games/M/mindup.ts [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const mindup: GamePresence = {
+	logo: "https://i.imgur.com/M3Y4LQu.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/M/mindup.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/M/mrjack.ts [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const mrjack: GamePresence = {
+	logo: "https://i.imgur.com/FTg4JiZ.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/M/mrjack.ts:13]: Check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
K, L, and M folders

File: websites/B/Board Game Arena/games/K/keyflower.ts [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const keyflower: GamePresence = {
+	logo: "https://i.imgur.com/nfSbrAw.png",
```
[the remark this answers is not in the document]
Comment (theusaf) [reply websites/B/Board Game Arena/games/K/keyflower.ts:13]: image

File: websites/B/Board Game Arena/games/N/nidavellir.ts [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const nidavellir: GamePresence = {
+	logo: "https://i.imgur.com/3n8YF6Y.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/N/nidavellir.ts:13]: Check light theme

File: websites/B/Board Game Arena/games/N/northwestpassage.ts [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const northwestpassage: GamePresence = {
+	logo: "https://i.imgur.com/QS7Tlt6.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/N/northwestpassage.ts:13]: Check light theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
N, O, P and Q folders

**Review by Bas950 — COMMENTED** [review]
U and V folders checked
Only R, S and T left
image

File: websites/B/Board Game Arena/games/R/regicide.ts [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const regicide: GamePresence = {
+	logo: "https://i.imgur.com/g40uvg4.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/R/regicide.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/R/regidice.ts [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const regidice: GamePresence = {
+	logo: "https://i.imgur.com/EK0QUsf.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/R/regidice.ts:13]: Check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
R folder

File: websites/B/Board Game Arena/games/S/shiftingstones.ts [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const shiftingstones: GamePresence = {
+	logo: "https://i.imgur.com/rQU4MQu.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/S/shiftingstones.ts:13]: Check white theme

File: websites/B/Board Game Arena/games/S/spiritsoftheforest.ts [hunk 30]
```diff
@@ -0,0 +1,43 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const spiritsoftheforest: GamePresence = {
+	logo: "https://i.imgur.com/pXGb1hO.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/S/spiritsoftheforest.ts:13]: Check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
S folder

File: websites/B/Board Game Arena/games/S/shiftingstones.ts (hunk 29 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/S/shiftingstones.ts:13]: image

File: websites/B/Board Game Arena/games/T/tiwanaku.ts [hunk 31]
```diff
@@ -0,0 +1,35 @@
...
+	getCurrentGameState,
+	getCurrentGameStateType,
+	getPlayerAvatar,
+	getPlayerData,
+	getPlayerScore,
+	getUserPlayerId,
+} from "../../util";
+
+const tiwanaku: GamePresence = {
+	logo: "https://i.imgur.com/xSNN8UW.png",
```
Comment (Bas950) [inline websites/B/Board Game Arena/games/T/tiwanaku.ts:13]: Check white theme

**Review by Bas950 — CHANGES_REQUESTED** [review]
T folder

File: websites/B/Board Game Arena/games/T/tiwanaku.ts (hunk 31 above)
Comment (theusaf) [reply websites/B/Board Game Arena/games/T/tiwanaku.ts:13]: image

**Review by Bas950 — APPROVED** [review]

**Pull request merged**